cattle/agent/handler.py

```diff
@@ -89,7 +89,7 @@ def _get_response_data(self, obj):
                 pass
 
             if method is not None:
-                return method(obj)
+                return {type: method(obj)}
```

Comment:
You need to grab the type from the resourceType field of the request.  This is because the type field of the object may not match the resourceType.  For example, a container will have type='container' but the resourceType='instance'.  This is because the backend orchestration never deals with subtypes really.  If you look at ObjectManager.getType() it will get the parent most type of the object.